elastic/kibana
# [Console] Move out of legacy + migrate server side to New Platform

## Summary

Continuation of https://github.com/elastic/kibana/pull/43346. Move the `console` AND `console_extensions` to New Platform (NP).

## How to review

- A _lot_ of files have moved and a _lot_ of them are JSON. It is safe to ignore these.
- Files of importance to review are:
    - `src/legacy/core_plugins/console_legacy/**/*`
    - `src/plugins/console/public/*.ts`
    - `src/plugins/console/server/*.ts`

## Testing

Because server has been migrated to NP in this PR it will be important to test thoroughly, we need to make sure that the `kibana.yml` config works as expected and that the proxying behaviour works as expected including fallback to nodes.

Some things to run through (besides the Test Rail test case CC @cuff-links)

 - Check that autocomplete as expected works in x-pack and OSS
 - Issue a request that will error and ensure it works
 - Ensure we can open Docs as before and that copy to curl works.

## Outstanding/known issues:

- [x] Error responses in NP are currently wrapped in `Boom`. This does not work with Console's proxying behaviour. This PR updates this behaviour in `src/core/server/http/router/response_adapter.ts`. This needs independent review as it is a platform core update.
- [x] There is a known issue with config parsing at the moment https://github.com/elastic/kibana/issues/55576. I might be doing something wrong though. [UPDATE] this was fixed in https://github.com/elastic/kibana/pull/56105.


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.
